mp/src/game/client/neo/ui/neo_root.cpp

```diff
@@ -0,0 +1,2068 @@
+#include "tier0/memdbgon.h"
+
+// TODO: Gamepad
+//   Gamepad enable: joystick 0/1
+//   Reverse up-down axis: joy_inverty 0/1
+//   Swap sticks on dual-stick controllers: joy_movement_stick 0/1
+//   Horizontal sens: joy_yawsensitivity: -0.5 to -7.0
+//   Vertical sens: joy_pitchsensitivity: 0.5 to 7.0
+
+using namespace vgui;
+
+// See interface.h/.cpp for specifics:  basically this ensures that we actually Sys_UnloadModule
+// the dll and that we don't call Sys_LoadModule over and over again.
+static CDllDemandLoader g_GameUIDLL( "GameUI" );
+
+CNeoRoot *g_pNeoRoot = nullptr;
+void NeoToggleconsole();
+
+namespace {
+
+inline NeoUI::Context g_uiCtx;
+int g_iRowsInScreen;
+int g_iAvatar = 64;
+int g_iRootSubPanelWide = 600;
+int g_iGSIX[GSIW__TOTAL] = {};
+constexpr wchar_t WSZ_GAME_TITLE[] = L"neatbkyoc ue";
+
+const wchar_t *QUALITY_LABELS[] = {
+	L"Low",
+	L"Medium",
+	L"High",
+	L"Very High",
+};
+
+enum QualityEnum
+{
+	QUALITY_LOW = 0,
+	QUALITY_MEDIUM,
+	QUALITY_HIGH,
+	QUALITY_VERYHIGH,
+};
+
+ConCommand neo_toggleconsole("neo_toggleconsole", NeoToggleconsole);
+}
+
+void OverrideGameUI()
+{
+	if (!OverrideUI->GetPanel())
+	{
+		OverrideUI->Create(0U);
+	}
+
+	if (g_pNeoRoot->GetGameUI())
+	{
+		g_pNeoRoot->GetGameUI()->SetMainMenuOverride(g_pNeoRoot->GetVPanel());
+	}
+}
+
+extern ConVar neo_fov;
+extern ConVar neo_viewmodel_fov_offset;
+extern ConVar neo_aim_hold;
+extern ConVar cl_autoreload_when_empty;
+extern ConVar cl_righthand;
+extern ConVar cl_showpos;
+extern ConVar cl_showfps;
+extern ConVar hud_fastswitch;
+
+enum MultiThreadIdx
+{
+	THREAD_SINGLE = 0,
+	THREAD_MULTI,
+};
+
+enum FilteringEnum
+{
+	FILTERING_BILINEAR = 0,
+	FILTERING_TRILINEAR,
+	FILTERING_ANISO2X,
+	FILTERING_ANISO4X,
+	FILTERING_ANISO8X,
+	FILTERING_ANISO16X,
+
+	FILTERING__TOTAL,
+};
+
+/////////////////
+// SERVER BROWSER
+/////////////////
+
+enum AnitCheatMode
+{
+	ANTICHEAT_ANY,
+	ANTICHEAT_ON,
+	ANTICHEAT_OFF,
+
+	ANTICHEAT__TOTAL,
+};
+
+static const wchar_t *ANTICHEAT_LABELS[ANTICHEAT__TOTAL] = {
+	L"<Any>", L"On", L"Off"
+};
+
+void CNeoServerList::UpdateFilteredList()
+{
+	// TODO: g_pNeoRoot->m_iSelectedServer Select kept with sorting
+	m_filteredServers = m_servers;
+	if (m_filteredServers.IsEmpty())
+	{
+		return;
+	}
+
+	// Can't use lamda capture for this, so pass through context
+	V_qsort_s(m_filteredServers.Base(), m_filteredServers.Size(), sizeof(gameserveritem_t),
+			  [](void *vpCtx, const void *vpLeft, const void *vpRight) -> int {
+		const GameServerSortContext gsCtx = *(static_cast<GameServerSortContext *>(vpCtx));
+		auto *pGsiLeft = static_cast<const gameserveritem_t *>(vpLeft);
+		auto *pGsiRight = static_cast<const gameserveritem_t *>(vpRight);
+
+		// Always set szLeft/szRight to name as fallback
+		const char *szLeft = pGsiLeft->GetName();
+		const char *szRight = pGsiRight->GetName();
+		int iLeft, iRight;
+		bool bLeft, bRight;
+		switch (gsCtx.col)
+		{
+		break; case GSIW_LOCKED:
```

Comment:
What a weird style lol with this `break;` in front of `case`, can we use the "normal" one as in the next `switch` case on the line 189?